ui/pages/send/gas-display/gas-display.js

```diff
@@ -0,0 +1,367 @@
+import GasTiming from '../../../components/app/gas-timing/gas-timing.component';
+import InfoTooltip from '../../../components/ui/info-tooltip/info-tooltip';
+import LedgerInstructionField from '../../../components/app/ledger-instruction-field/ledger-instruction-field';
+import TextField from '../../../components/ui/text-field';
+import Typography from '../../../components/ui/typography/typography';
+import Button from '../../../components/ui/button';
+import Box from '../../../components/ui/box';
+import { TYPOGRAPHY } from '../../../helpers/constants/design-system';
+import LoadingHeartBeat from '../../../components/ui/loading-heartbeat';
+import TransactionDetailItem from '../../../components/app/transaction-detail-item/transaction-detail-item.component';
+import { NETWORK_TO_NAME_MAP } from '../../../../shared/constants/network';
+import TransactionDetail from '../../../components/app/transaction-detail/transaction-detail.component';
+import ActionableMessage from '../../../components/ui/actionable-message/actionable-message';
+import DepositPopover from '../../../components/app/deposit-popover/deposit-popover';
+
+const renderHeartBeatIfNotInTest = () =>
+  process.env.IN_TEST ? null : <LoadingHeartBeat />;
+export default function GasDisplay({
+  primaryTotalTextOverride,
+  secondaryTotalTextOverride,
+  hexTransactionTotal,
+  useNonceField,
+  customNonceValue,
+  updateCustomNonce,
+  nextNonce,
+  getNextNonce,
+  draftTransaction,
+  useNativeCurrencyAsPrimaryCurrency,
+  primaryTotalTextOverrideMaxAmount,
+  maxFeePerGas,
+  maxPriorityFeePerGas,
+  isMainnet,
+  showLedgerSteps,
+  isBuyableChain,
+  hexMinimumTransactionFee,
+  hexMaximumTransactionFee,
+  nativeCurrency,
+  chainId,
+  showAccountDetails,
+  gasError,
+}) {
+  const t = useContext(I18nContext);
+  const { estimateUsed, supportsEIP1559V2 } = useGasFeeContext();
+  const [showDepositPopover, setShowDepositPopover] = useState(false);
+  const networkName = NETWORK_TO_NAME_MAP[chainId];
+  const nonceField = useNonceField ? (
+    <div>
+      <div className="confirm-detail-row">
+        <div className="confirm-detail-row__label">
+          {t('nonceFieldHeading')}
+        </div>
+        <div className="custom-nonce-input">
+          <TextField
+            type="number"
+            min="0"
+            placeholder={
+              typeof nextNonce === 'number' ? nextNonce.toString() : null
+            }
+            onChange={({ target: { value } }) => {
+              if (!value.length || Number(value) < 0) {
+                updateCustomNonce('');
+              } else {
+                updateCustomNonce(String(Math.floor(value)));
+              }
+              getNextNonce();
+            }}
+            fullWidth
+            margin="dense"
+            value={customNonceValue || ''}
+          />
+        </div>
+      </div>
+    </div>
+  ) : null;
+  const renderTotalMaxAmount = () => {
```

Comment:
Why is there a nonce field in the gas-display component?

Reply:
@brad-decker You are right, a nonce field doesn't need to be in gas-display component. I deleted it [here](https://github.com/MetaMask/metamask-extension/pull/14634/commits/71c1459bb901b933d843ba0af7206893f7c988f8).